### PR TITLE
fix: Serverless Deployment provider packs correctly 

### PR DESCRIPTION
**Pull request by jwwisgerhof**
Fixes redwoodjs/redwood#10528

Simple fix that restores Serverless deployment functionality.

### CODE REVIEW

**Comment by dthyresson** [comment]
Thanks @jwwisgerhof for fixing! I don't have a serverless deployed app to test this. Could you provide the logs are some evidence of this fixing working (redacted for any sensitive info) so we can review approve and merge.

**Comment by jwwisgerhof** [comment]
Hard to provide logs that show a definite fix, but have tested this on our enterprise app and am using it live at the moment. WIthout fix:

```
[STARTED] Checking if Serverless framework is installed...
Framework Core: 3.38.0 (local)
Plugin: 7.2.0
SDK: 4.5.1

[COMPLETED] Checking if Serverless framework is installed...
[STARTED] Building api...
❯ Generating Prisma Client...
✔ Generating Prisma Client...
❯ Verifying graphql schema...
✔ Verifying graphql schema...
❯ Building API...
✔ Building API...
[COMPLETED] Building api...
[STARTED] Packing Functions...
[FAILED] undefined is not a function
undefined is not a function
```

With my branch built and lifted into our project:
```
[STARTED] Checking if Serverless framework is installed...
Framework Core: 3.38.0 (local)
Plugin: 7.2.0
SDK: 4.5.1

[COMPLETED] Checking if Serverless framework is installed...
[STARTED] Building api...
❯ Generating Prisma Client...
✔ Generating Prisma Client...
❯ Verifying graphql schema...
✔ Verifying graphql schema...
❯ Building API...
✔ Building API...
[COMPLETED] Building api...
[STARTED] Packing Functions...
[COMPLETED] Packing Functions...
[STARTED] Deploying api....
[SKIPPED] Finishing early due to --pack-only flag. Your Redwood project is packaged and ready to deploy
```

I'll set up a test repo with minimal code and link it here.

**Comment by jwwisgerhof** [comment]
Alright - gutted our repo quickly, probably a bit messy but it serves the purpose. 
Working repo on 6.4: https://github.com/jwwisgerhof/rw-75-sls-test
Upgraded repo to 7.5.1: https://github.com/jwwisgerhof/rw-75-sls-test/tree/RW-75

To reproduce in either repo:
- `yarn install`
- `yarn rw deploy sls --stage uat --pack-only --side api`

Succeeds on main, fails on RW-75 branch. Build my PR and copy dist over to RW-75 branch and deploy command succeeds.

Let me know if you need more info :)

File: packages/cli/src/commands/deploy/packing/nft.js
Comment (thedavidprice) [inline packages/cli/src/commands/deploy/packing/nft.js:10]: Do we have any idea about what changed and broke this in the first place? This is a surprising fix given that the error showed up after a minor release... Assuming the culprit could be with the NFT package itself. 🤷‍♂️

No need to hold up the PR to try and figure it out — goal here is simple + efficient. Just thought it was worth asking.

Comment (jwwisgerhof) [reply packages/cli/src/commands/deploy/packing/nft.js:10]: I am not 100% sure - I got to this fix by simple trial and error, realizing eventually that "exports" is undefined. Having a look around the web now , it has to do with how the JS is compiled / emitted. Modern versions don't allow the `exports.xxx` syntax depending on module resolution.

I am not familiar enough with the RW codebase to see what change might have caused this. I did go through our build log and can confirm that in `@redwoodjs/cli@npm:7.2.0-rc.53` this still worked and packed correctly.

**Review by dthyresson — APPROVED** [review]
Very much appreciated!

**Comment by dthyresson** [comment]
@jwwisgerhof One last thing before I can merge. Might you add a changeset so the details of this fix gets into the release notes?

See: https://github.com/redwoodjs/redwood/blob/main/.changesets/10668.md for examples.

Appreciate it!

File: packages/cli/src/commands/deploy/packing/nft.js
Comment (thedavidprice) [reply packages/cli/src/commands/deploy/packing/nft.js:10]: > it has to do with how the JS is compiled / emitted

Yeah, this is the kind of thing that has me thinking a bit; one of our related build dependencies could have been updated behind the scenes and introduced a behavior change that's breaking. 

Again, not a critical priority to figure out. Appreciate you taking time to reply 👍

**Comment by jwwisgerhof** [comment]
Thanks @dthyresson - have added that now. Looking forward to getting this released and upgrading to RW7!

**Pull request merged**
